presto-main/src/main/java/com/facebook/presto/execution/scheduler/LegacySqlQueryScheduler.java

```diff
@@ -190,6 +196,7 @@ private LegacySqlQueryScheduler(
         this.queryStateMachine = requireNonNull(queryStateMachine, "queryStateMachine is null");
         this.plan.compareAndSet(null, requireNonNull(plan, "plan is null"));
         this.session = requireNonNull(session, "session is null");
+        this.metadata = requireNonNull(metadata, "metadata is null");
```

Comment:
nit: just get the functionmanager from the metadata since that's all you need (and same below)